src/ExportMap.js

```diff
@@ -782,7 +782,7 @@ let prevSettings = '';
  * also calculate a cacheKey, where parts of the cacheKey hash are memoized
  */
 function childContext(path, context) {
-  const { settings, parserOptions, parserPath } = context;
+  const { settings, parserOptions, parserPath, filename } = context;
```

Comment:
https://eslint.org/blog/2023/05/eslint-v8.40.0-released/#highlights

1. `physicalFilename` should be preferred over `filename` and fallback it when `physicalFilename` is unavailable
2. `filename/physicalFilename` are not available on ESLint 7